src/world_new/World.cpp

```diff
@@ -80,5 +80,12 @@ uint64_t World::getTimeDifference() const noexcept { return tickDuration; }
 
 robot::RobotControllers &World::getControllersForRobot(uint8_t id) noexcept { return robotControllers[id]; }
 
-size_t World::getHistorySize() const noexcept { return history.size(); }
+size_t World::getHistorySize() const noexcept {
+    return history.size();
+}
+
+Settings World::getSettings() const noexcept {
+    return Settings(*settings);
+}
```

Comment:
While I get that you need it for tests, this is one of those ugly things that will propagate through our system with people using it everywhere and then we have no seperation of concerns anymore. Get it at the source, please no more hookups to World which are unnecessary (e.g. controllers), especially not for testing purposes.

Reply:
OK but i need to access the settings. I need to restructure some things in the test environment then. I do agree with you however.